Automattic/woocommerce-payments
# Handle a new partly restricted account status

#### Handle partly restricted account status

New account status has been introduced that tells the end customer that some payment methods or deposits has been disabled. This PR handles the new status and show correct message to the end user.

<img width="1104" alt="Screenshot 2022-05-20 at 17 39 22" src="https://user-images.githubusercontent.com/5311119/169551784-37e3714e-5509-40fd-b97e-da368a07774e.png">

#### Testing instructions

- Make sure that you one of the payment methods doesn't have all necessary documents. The best way to achieve this is to submit for new payment method and stop it at the middle of setup.
- On your ~status~ overview page you should see a message that says that some payment methods or deposits has been disabled.

If you'd like to simulate the new status, you can apply this snippet locally:

```diff
diff --git a/includes/class-wc-payments-account.php b/includes/class-wc-payments-account.php
index 196858fb7..a39b50e36 100644
--- a/includes/class-wc-payments-account.php
+++ b/includes/class-wc-payments-account.php
@@ -186,6 +186,8 @@ class WC_Payments_Account {
                        ];
                }
 
+               $account['status'] = 'restricted_partially';
+
                return [
                        'email'               => $account['email'] ?? '',
                        'country'             => $account['country'] ?? 'US',
```

-------------------
